tools/codegen/src/main.rs

```diff
@@ -82,17 +84,37 @@ fn main() -> Result<()> {
         .collect();
 
     let mut crates_io_info = None;
+    let mut crates_io_version_detail = None;
     base_info.rust_crate = base_info
         .rust_crate
         .as_ref()
         .map(|s| replace_vars(s, package, None, None, base_info.rust_crate.as_deref()))
         .transpose()?;
     if let Some(crate_name) = &base_info.rust_crate {
         eprintln!("downloading crate info from https://crates.io/api/v1/crates/{crate_name}");
-        crates_io_info = Some(
-            download(&format!("https://crates.io/api/v1/crates/{crate_name}"))?
-                .into_json::<crates_io::Crate>()?,
+        let info = download(&format!("https://crates.io/api/v1/crates/{crate_name}"))?
+            .into_json::<crates_io::Crate>()?;
+        let latest_version = &info.versions[0].num;
+        crates_io_version_detail = Some(
+            download(&format!("https://crates.io/api/v1/crates/{crate_name}/{latest_version}"))?
+                .into_json::<crates_io::VersionMetadata>()?
+                .version,
         );
+
+        if let Some(crate_repository) = info.crate_.repository.clone() {
+            // cargo-dinghy being fixed at https://github.com/sonos/dinghy/pull/231
+            if crate_name != "cargo-dinghy"
+                && !crate_repository
+                    .to_lowercase()
+                    .starts_with(&base_info.repository.to_lowercase())
+            {
+                panic!("warning: crate metadata repository {crate_repository} differs from base manifest");
+            }
```

Comment:
need to remove "warning: " since it is panicing

The rationale for panic-ing in this case is https://github.com/taiki-e/install-action/issues/488 .  If the repo has moved, that should require human intervention.